test/coverage/CMakeLists.txt

```diff
@@ -14,59 +14,93 @@ if (CMAKE_BUILD_TYPE AND NOT ("${CMAKE_BUILD_TYPE}" STREQUAL "Debug"))
+    message (FATAL_ERROR "Coverage tests needs at least GCC8.")
+endif ()
+
+find_program (GCOVR_COMMAND NAMES gcovr)
 
-if (NOT LCOV_COMMAND)
-    message(FATAL_ERROR "lcov not found! Aborting...")
+if (NOT GCOVR_COMMAND)
+    message(FATAL_ERROR "gcovr not found! Aborting...")
 endif ()
 
-# Files which should not be included in the coverage report
-set (TEST_COVERAGE_EXCLUDE_FILES
-    "'/usr/*'"
-    "'${SEQAN3_CLONE_DIR}/include/seqan3/contrib/*'"
-    "'${SEQAN3_CLONE_DIR}/include/seqan3/std/*'"
-    "'${SEQAN3_CLONE_DIR}/submodules/*'"
-    "'${SEQAN3_CLONE_DIR}/test/unit/*'"
-    "'${PROJECT_BINARY_DIR}/vendor/*'"
-)
 # Holds all target's defined by seqan3_test
 set_property (GLOBAL PROPERTY GLOBAL_TEST_COVERAGE_ALL_TESTS "")
 
-add_custom_command (
-    OUTPUT ${PROJECT_BINARY_DIR}/seqan3_coverage
-    # Cleanup lcov (resetting code coverage counters to zero)
-    COMMAND ${LCOV_COMMAND} --directory ${PROJECT_BINARY_DIR} --zerocounters
-    # Create baseline to make sure untouched files show up in the report
-    COMMAND ${LCOV_COMMAND} --directory ${PROJECT_BINARY_DIR} --capture --initial --output-file ${PROJECT_BINARY_DIR}/seqan3_coverage.baseline
+set (SEQAN3_COVERAGE_PARALLEL_LEVEL "1" CACHE STRING "Number of threads to use for coverage report generation.")
+
+# Arguments for gcovr.
+# Each line is an argument, ';' will end up being a whitespace due to expansion by CMake.
+# Special characters like '"', '\', ')' need to be escaped by a number of '\', depending how often they are evaluated.
+# gcovr uses python regex expressions.
+# See https://gcovr.com/en/5.0/guide.html#the-gcovr-command for an overview of parameters.
+# See https://gcovr.com/en/5.0/faq.html#why-does-c-code-have-so-many-uncovered-branches for an explanation on branches.
+set (SEQAN3_GCOVR_ARGUMENTS "")
+# The directory of the CMakeLists.txt used for invoking cmake.
+list (APPEND SEQAN3_GCOVR_ARGUMENTS "--root;${CMAKE_CURRENT_LIST_DIR}")
+# The build directory.
+list (APPEND SEQAN3_GCOVR_ARGUMENTS "${PROJECT_BINARY_DIR}")
+# Include all files whose path match '${SEQAN3_CLONE_DIR}/include/seqan3/.*'.
+list (APPEND SEQAN3_GCOVR_ARGUMENTS "--filter;\'${SEQAN3_CLONE_DIR}/include/seqan3/.*\'")
+# Include all files whose path match '${SEQAN3_CLONE_DIR}/test/include/seqan3/test/.*'.
+list (APPEND SEQAN3_GCOVR_ARGUMENTS "--filter;\'${SEQAN3_CLONE_DIR}/test/include/seqan3/test/.*\'")
+# Remove all files whose path match '${SEQAN3_CLONE_DIR}/include/seqan3/contrib/.*'.
+list (APPEND SEQAN3_GCOVR_ARGUMENTS "--exclude;\'${SEQAN3_CLONE_DIR}/include/seqan3/contrib/.*\'")
+# Remove all files whose path match '${SEQAN3_CLONE_DIR}/include/seqan3/std/.*'.
+list (APPEND SEQAN3_GCOVR_ARGUMENTS "--exclude;\'${SEQAN3_CLONE_DIR}/include/seqan3/std/.*\'")
+# Remove line coverage for all lines that match '^[\s]*$', i.e. empty lines.
+list (APPEND SEQAN3_GCOVR_ARGUMENTS "--exclude-lines-by-pattern;\\\'^\\\\s*$$\\\'")
```

Comment:
Maybe we can write an escape command for this? Also, `^[\s]*$` doesn't match the escaped command `'^\s*$'`, or does it?

Did you try using VERBATIM in add_custom_command? https://cmake.org/cmake/help/latest/command/add_custom_command.html#add-custom-command ?

Reply:
I did, but it then had trouble when calling the bash script :/

Escape command sounds good, maybe it would also be possible to use configure_file on the gcov.sh ...